src/libstore/s3_binary_cache_store.go

```diff
@@ -26,46 +25,30 @@ func (s *S3Store) getUri() string {
 }
 
 func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
-	found, err := s.client.BucketExists(ctx.Background(), s.bucketName)
-	if err != nil {
-		fmt.Println(err)
-		return
-	}
-	if found {
-		obj, err := s.client.GetObject(ctx, s.bucketName, path, minio.GetObjectOptions{})
-		if err != nil {
-			log.Fatalln(err)
-			return
-		}
-		defer reader.Close()
-
-		localFile, err := os.Create("my-testfile")
+	// found, err := s.client.BucketExists(ctx, s.bucketName)
+	// if err != nil {
+	// 	fmt.Println(err)
+	// 	return false, err
+	// }
+
+	// if found {
+	// 	fmt.Println(found)
+		_, err := s.GetFile(ctx, path)
 		if err != nil {
-			log.Fatalln(err)
+			return false, err
 		}
-		defer localFile.Close()
-
-		stat, err := reader.Stat()
-		if err != nil {
-			log.Fatalln(err)
-		}
-
-		if _, err := io.CopyN(localFile, reader, stat.Size); err != nil {
-			log.Fatalln(err)
-			return
-		}
-	}
-
-	return false
+		// defer obj.Close()
+	// }
+	return true, err
```

Comment:
make it more obvious that err is always nil in that branch
```suggestion
	return true, nil
```